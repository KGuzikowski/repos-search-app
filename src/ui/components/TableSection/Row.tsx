import React from 'react'
import { repoType } from '../../../redux/repos/repo.types'
import { Td } from './TableSection.styles'

type RowProps = {
    repo: repoType,
    highlighted: boolean
}

const Row = ({ repo, highlighted }: RowProps) => (
    <tr>
        <Td highlighted={highlighted}>{repo.id}</Td>
        <Td highlighted={highlighted}>{repo.name}</Td>
        <Td highlighted={highlighted}>{repo.owner.name}</Td>
        <Td highlighted={highlighted}>{repo.stars}</Td>
        <Td highlighted={highlighted}>{repo.createdAt}</Td>
    </tr>
)

export default Row 