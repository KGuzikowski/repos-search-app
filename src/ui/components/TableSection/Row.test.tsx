import React from 'react'
import { create } from 'react-test-renderer'
import Row from './Row'

it('renders Row component correctly', () => {
    const repo = {
        id: 1,
        name: 'repo',
        owner: { id: 1, name: 'Karol' },
        stars: 2,
        createdAt: '2019-05-12'
    }
    const component = create(<Row repo={repo} highlighted />)
    expect(component.toJSON()).toMatchSnapshot()
})