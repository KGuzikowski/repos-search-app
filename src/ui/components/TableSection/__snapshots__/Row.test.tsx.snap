// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Row component correctly 1`] = `
<tr>
  <td
    className="sc-AxheI gsTZvP"
  >
    1
  </td>
  <td
    className="sc-AxheI gsTZvP"
  >
    repo
  </td>
  <td
    className="sc-AxheI gsTZvP"
  >
    Karol
  </td>
  <td
    className="sc-AxheI gsTZvP"
  >
    2
  </td>
  <td
    className="sc-AxheI gsTZvP"
  >
    2019-05-12
  </td>
</tr>
`;
